src/signature.c

```diff
@@ -93,14 +93,10 @@ gboolean signature_init(GError **error)
 		return FALSE;
 	}
 
-	id = X509_PURPOSE_get_count() + 1;
-	if (X509_PURPOSE_get_by_id(id) >= 0) {
-		g_set_error_literal(
-				error,
-				R_SIGNATURE_ERROR,
-				R_SIGNATURE_ERROR_CRYPTOINIT_FAILED,
-				"Failed to calculate free OpenSSL X509 purpose id");
-		return FALSE;
+	/* OpenSSL 3.5 warns that there may be gaps, so we need to search. */
```

Comment:
Could we use `X509_PURPOSE_get_unused_id()` optionally with `#if OPENSSL_VERSION_PREREQ(3, 0)` ?

Reply:
I've added a comment.